src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -186,13 +187,13 @@ def handle_sso_command(cmd):
             # pylint: disable=redefined-builtin
             FileNotFoundError = IOError
         try:
-            f = open(sp_x_509_cert, 'r')
+            f = open(sp_x_509_cert, 'r', encoding='utf-8') if six.PY3 else open(sp_x_509_cert, 'rb')
             sp_x_509_cert = f.read()
             f.close()
         except FileNotFoundError:
             pass
         try:
-            f = open(sp_private_key, 'r')
+            f = open(sp_private_key, 'r', encoding='utf-8') if six.PY3 else open(sp_private_key, 'rb')
```

Comment:
I understand you don't want to change this too much compared to master/octopus, but I wanted to bring in the context manager form:
```py
with open() as f:
  sp_private_key = f.read()
# no need to explicitly close
```

Reply:
Provided it's a bugfix, I prefer to introduce the minimal changes in code structure.